aiida/engine/processes/workchains/workchain.py

```diff
@@ -121,24 +121,53 @@ def on_run(self):
         super().on_run()
         self.node.set_stepper_state_info(str(self._stepper))
 
+    def _resolve_nested_context(self, key: str) -> Tuple[AttributeDict, str]:
+        """
+        Returns a reference to a sub-dictionary of the context and the last key,
+        after resolving a potentially segmented key where required sub-dictionaries are created as needed.
+
+        :param key: A key into the context, where words before a dot are interpreted as a key for a sub-dictionary
+        """
+        ctx = self.ctx
+        ctx_path = key.split('.')
+
+        for path in ctx_path[:-1]:
+            try:
+                ctx = ctx[path]
+            except TypeError:
+                raise AssertionError(
+                    f'The sub-dictionary part `{path}` of the given nested key path `{key}` is already set to a list'
```

Comment:
The part `path` here will contain just the immediate part of the full path that causes problems. It would be good if we could provide the full sub path. This will prevent ambiguity if you have a path with sub paths that have the same name, for example `some.name.name.space`.

Reply:
Agreed, will fix.

Reply:
Wait, `{key}` is that info, but I will reformat the message to follow "problem: reason" to make it more clear, including `{key}` in the "problem" part.

Reply:
Sure `key` will show that `some.name.name.space` is a problematic key but it will say that it is because

> The sub-dictionary part `name` ... is already set to a list

The point is that then you don't know which `name` part is the problematic one, the first or the second. If you give the full path, i.e., `some.name` or `some.name.name`, the error is unambiguous